docs/usage/customsessions.md

```diff
@@ -0,0 +1,145 @@
+# Create a `CustomSessionStorage` Solution
+
+This library comes with two session management options: `MemorySessionStorage` and `CustomSessionStorage`.
+
+`MemorySessionStorage` exists as an option to help you get started developing your apps as quickly as possible, and is the default storage option on `Shopify.Context`. It's perfect for working in your development and testing environments. However, this storage solution is not meant to be used in production due to its limitations.
+
+When you're ready to deploy your app and run it in production, you'll need to set up a `CustomSessionStorage`, which you can then use in initializing your `Shopify.Context`. The `CustomSessionStorage` class expects to be initialized with three callbacks that link to your chosen storage solution and map to the `storeSession`, `loadSession`, and `deleteSession` methods on the class.
+
+## Callback Methods
+
+- All of the callbacks used to create a new instance of `CustomSessionStorage` should be `async` functions and return a `Promise` that resolves to a specified type, as outlined below.
+
+| Method           | Arg Type  | Return Type                                                 | Notes                                                                                                                                                                                                                |
+| ---------------- | --------- | ----------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `storeCallback`  | `Session` | `Promise<boolean>`                                           | Takes in the `Session` to be stored, returns a `boolean` (`true` if stored successfully).                                                                                                                            |
+| `loadCallback`   | `string`  | `Promise<Session \| Record<string, unknown> \| undefined> ` | Takes in the id of the `Session` to load (as a `string`) and returns either an instance of a `Session`, an object to be used to instantiate a `Session`, or `undefined` if no record is found for the specified id. |
+| `deleteCallback` | `string`  | `Promise<boolean>`                                           | Takes in the id of the `Session` to load (as a `string`) and returns a  `booelan` (`true` if deleted successfully).                                                                                                  |
+
+## Example Usage
+
+This is an example implementation of a `CustomSessionStorage` solution, using `redis` for storage.
+
+### Dependencies
+Before starting this tutorial, please first follow our [Getting Started Guide](../getting_started.md).
+Once completed, you will need to add the `redis` package to your project by running:
+`$ yarn add redis`
```

Comment:
Scanning this I initially missed the bit about installing redis because the yarn command was inline; I think a good rule would be that `yarn add` commands should get block treatment for higher visibility:

----

Before starting this tutorial, please first follow our [Getting Started Guide](../getting_started.md).

Once completed, you'll need to add the `redis` package to your project:

```sh
$ yarn add redis
```